modules/tensor_mechanics/src/materials/ComputeStressBase.C

```diff
@@ -44,46 +44,70 @@ ComputeStressBase::ComputeStressBase(const InputParameters & parameters)
     _elasticity_tensor(getMaterialPropertyByName<RankFourTensor>(_elasticity_tensor_name)),
     _extra_stress(getDefaultMaterialProperty<RankTwoTensor>(_base_name + "extra_stress")),
     _Jacobian_mult(declareProperty<RankFourTensor>(_base_name + "Jacobian_mult")),
-    _store_stress_old(getParam<bool>("store_stress_old"))
+    _store_stress_old(getParam<bool>("store_stress_old")),
+    // InitialStress TODO: remove the following
+    _initial_stress_provided(getParam<std::vector<FunctionName>>("initial_stress").size() ==
+                             LIBMESH_DIM * LIBMESH_DIM),
+    _initial_stress(_initial_stress_provided
+                        ? &declareProperty<RankTwoTensor>(_base_name + "initial_stress")
+                        : nullptr),
+    _initial_stress_old(_initial_stress_provided
+                            ? &getMaterialPropertyOld<RankTwoTensor>(_base_name + "initial_stress")
+                            : nullptr)
 {
-  // Declares old stress and older stress if the parameter _store_stress_old is true. This parameter
-  // can be set from the input file using any of the child classes of ComputeStressBase.
 
+  if (getParam<bool>("use_displaced_mesh"))
+    mooseError("The stress calculator needs to run on the undisplaced mesh.");
+
+  // InitialStress TODO: remove following initial stress stuff
   const std::vector<FunctionName> & fcn_names(
       getParam<std::vector<FunctionName>>("initial_stress"));
   const unsigned num = fcn_names.size();
 
-  if (!(num == 0 || num == 3 * 3))
+  if (!(num == 0 || num == LIBMESH_DIM * LIBMESH_DIM))
     mooseError("Either zero or ",
-               3 * 3,
+               LIBMESH_DIM * LIBMESH_DIM,
```

Comment:
Yea, i wonder how many other "3"s there are hanging around!

Reply:
This is a discussion for another time, but I think `LIBMESH_DIM` is horrible to use for 3, because it can be set to 1, 2, or 3 at compile time.  We need some other tensor size thing that is always 3. We'll just do a big search/replace on `LIBMESH_DIM` if we ever get around to it, so this is fine for now.